packages/web3-providers-http/types.ts

```diff
@@ -0,0 +1,28 @@
+export interface HttpRpcOptions {
+    id?: number,
+    jsonrpc: string,
+    method: string,
+    params: {[key: string]: string | number}
+}
+
+export interface HttpRpcResponse {
+    id: number,
+    jsonrpc: string,
+    result: string |
+        number |
+        boolean |
+        (string|number)[] |
+        {[key: string]: string | number} |
+        BigInt
+}
+
+export interface IWeb3Provider {
+    providerString: string
+    protectProvider: boolean
+    connected: boolean
+    supportsSubscriptions: boolean
+    setProvider: (providerString: string) => void
+    // TODO get rid of anys
+    send: (options: any) => Promise<any>
```

Comment:
Can't you just use `HttpRpcOptions`?

Reply:
`IWeb3Provider` should be in the `web3-providers-base` package, as any provider package should conform to this interface, but `options` could be anything depending on how the provider package works. We could do `HttpRpcOptins | WsRpcOptions | IpcOptions| any` to be a little more explicit though

Reply:
ahh ok, well in that case, you could have something like:
```
interface IWeb3Provider {
    providerString: string
    protectProvider: boolean
    connected: boolean
    supportsSubscriptions: boolean
    setProvider: (providerString: string) => void
    // TODO get rid of anys
    send: (options: any) => Promise<any>
 }
 
interface IHttpProvider extends IWeb3Provider {
  // if needed custom logic
}

Reply:
I moved `IWeb3Provider` into `web3-providers-base/types`

Reply:
Didn't see you last comment when I made mine, `IHttpProvider` doesn't need to exist...probably